activerecord/lib/active_record/sanitization.rb

```diff
@@ -144,7 +144,7 @@ def sanitize_sql_like(string, escape_character = "\\")
       #   # => "name='foo''bar' and group_id='4'"
       def sanitize_sql_array(ary)
         statement, *values = ary
-        if values.first.is_a?(Hash) && statement =~ /:\w+/
```

Comment:
any clue why this was added in the first place?
do we need, also, to update the doc for the method?